build_runner/lib/src/asset_graph/node.dart

```diff
@@ -169,6 +166,41 @@ class GeneratedAssetNode extends AssetNode {
       'GeneratedAssetNode: $id generated from input $primaryInput.';
 }
 
+/// A typical generated node which was generated in a specific phase, and used a
+/// set of [globs] during its generation.
+class GeneratedForPhaseAssetNode extends GeneratedAssetNode {
```

Comment:
What are your thoughts on `OutputNode` and `PhaseOutputNode`?

Reply:
Depending on how the `BuildAction` stuff pans out ya I like these names.